### ISSUE TITLE
Installation warnings Rev3

### ISSUE THREAD

**Issue by MikeMcC399** [cypress-io/cypress-realworld-app#1508]
- Follows on from closed issue https://github.com/cypress-io/cypress-realworld-app/issues/1403

## Description

Running `yarn install` produces installation warnings. Resolved issues are ticked in the list below:

- [x] [@material-ui/core](https://www.npmjs.com/package/@material-ui/core) - deprecated
- [x] [@material-ui/icons@4.11.3](https://www.npmjs.com/package/@material-ui/icons)
- [x] [@material-ui/lab](https://www.npmjs.com/package/@material-ui/lab) - deprecated
- [x] [aws-amplify](https://www.npmjs.com/package/aws-amplify)
- [ ] [react-infinite-calendar@2.3.1](https://www.npmjs.com/package/react-infinite-calendar)
deprecated
- [x] [@cypress/code-coverage](https://www.npmjs.com/package/@cypress/code-coverage)
- [x] [express-graphql](https://www.npmjs.com/package/express-graphql) - deprecated

## Steps to reproduce

Clone repo and execute the following under Node.js ~~`18.16.1`~~ `20.11.0`:

```bash
git clean -x -d -f
yarn install
```

## Expectation

There should be no installation warnings produced by `yarn install`.

### Clean install

```text
$ node -v
v18.16.1

$ yarn
yarn install v1.22.19
[1/5] Validating package.json...
[2/5] Resolving packages...
[3/5] Fetching packages...
[4/5] Linking dependencies...
warning " > @material-ui/core@4.12.4" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @material-ui/core@4.12.4" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/styles@4.11.5" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/styles@4.11.5" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/system@4.12.2" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/system@4.12.2" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/utils@4.11.3" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning "@material-ui/core > @material-ui/utils@4.11.3" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @material-ui/icons@4.11.3" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @material-ui/icons@4.11.3" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning " > @material-ui/lab@4.0.0-alpha.61" has incorrect peer dependency "react@^16.8.0 || ^17.0.0".
warning " > @material-ui/lab@4.0.0-alpha.61" has incorrect peer dependency "react-dom@^16.8.0 || ^17.0.0".
warning "aws-amplify > @aws-amplify/core > react-native-url-polyfill@1.3.0" has unmet peer dependency "react-native@*".
warning "aws-amplify > @aws-amplify/analytics > @aws-sdk/client-firehose > @aws-sdk/middleware-retry > react-native-get-random-values@1.9.0" has unmet peer dependency "react-native@>=0.56".
warning " > react-infinite-calendar@2.3.1" has incorrect peer dependency "react@^15.3.0 || ^16.0.0-alpha".
warning "react-infinite-calendar > react-tiny-virtual-list@2.2.0" has incorrect peer dependency "react@15.x || 16.x".
warning "react-infinite-calendar > react-transition-group@1.2.1" has incorrect peer dependency "react@^15.0.0 || ^16.0.0".
warning "react-infinite-calendar > react-transition-group@1.2.1" has incorrect peer dependency "react-dom@^15.0.0 || ^16.0.0".
warning "react-infinite-calendar > recompose@0.22.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0".
warning "@cypress/code-coverage > @cypress/webpack-preprocessor@5.17.1" has unmet peer dependency "@babel/preset-env@^7.0.0".
warning "@cypress/code-coverage > @cypress/webpack-preprocessor@5.17.1" has unmet peer dependency "babel-loader@^8.0.2 || ^9".
warning " > express-graphql@0.12.0" has incorrect peer dependency "graphql@^14.7.0 || ^15.3.0".
warning Workspaces can only be enabled in private projects.
[5/5] Building fresh packages...
$ husky install && patch-package
husky - Git hooks installed
patch-package 7.0.2
Applying patches...
@material-ui/core@4.12.4 ✔
react-virtualized@9.22.5 ✔
Done in 461.33s.
```

### Related issues

* [Migrate away from express-graphql package](https://github.com/cypress-io/cypress-realworld-app/issues/1420)
* [Migrate to MUI v5](https://github.com/cypress-io/cypress-realworld-app/issues/1418)
* [Migrate away from react-infinite-calendar](https://github.com/cypress-io/cypress-realworld-app/issues/1419)

**Comment by MikeMcC399** [comment]
- See also PR https://github.com/cypress-io/cypress-realworld-app/pull/1513

**Comment by jennifer-shehane** [comment]
We'd be open to PRs around these updates.

**Comment by MikeMcC399** [comment]
@jennifer-shehane 

> We'd be open to PRs around these updates.

I don't have the skills to provide these updates, so there is no plan from my side to submit any PRs to resolve this issue. @astone123 was previously covering these points.

**Comment by astone123** [comment]
@MikeMcC399 unfortunately, I don't think we'll have the bandwidth to prioritize this any time soon. If I have some free time, I'll try to look into this but I wouldn't count on it. 

If you feel comfortable trying to fix these issues and open a partial PR, that might be a good starting point for us to work from, even if you can't resolve everything at once. Thanks

**Comment by MikeMcC399** [comment]
@astone123 

> unfortunately, I don't think we'll have the bandwidth to prioritize this any time soon. If I have some free time, I'll try to look into this but I wouldn't count on it.
> 
> If you feel comfortable trying to fix these issues and open a partial PR, that might be a good starting point for us to work from, even if you can't resolve everything at once. Thanks

- I will take a fresh look, perhaps after https://github.com/cypress-io/cypress-realworld-app/pull/1513 has been resolved, however when I previously looked at this set of issues my conclusion was that I couldn't handle it.

**Comment by MikeMcC399** [comment]
- PR #1513 has solved the aws warning. Thanks to @timheilman / @AtofStryker !

**Comment by MikeMcC399** [comment]
- PR #1521 resolved the `express-graphql` issue. Thanks to @timheilman 👍🏻 🚀 !

**Comment by MikeMcC399** [comment]
- The issues with `@material-ui/*` modules have been resolved through PR #1522. Once again, kudos goes to @timheilman for an amazing amount of successful work! 👍🏻 💪🏻

**Comment by MikeMcC399** [comment]
Closing this issue.

The remaining installation warnings are covered in:

- https://github.com/cypress-io/cypress-realworld-app/issues/1419
- #1573
